src/main/java/org/mockito/internal/invocation/InvocationMatcher.java

```diff
@@ -141,8 +141,12 @@ private void captureVarargsPart(Invocation invocation) {
         for (ArgumentMatcher m : uniqueMatcherSet(indexOfVararg)) {
             if (m instanceof CapturesArguments) {
                 Object rawArgument = invocation.getRawArguments()[indexOfVararg];
-                for (int i = 0; i < Array.getLength(rawArgument); i++) {
-                    ((CapturesArguments) m).captureFrom(Array.get(rawArgument, i));
+                if (rawArgument != null) {
+                    for (int i = 0; i < Array.getLength(rawArgument); i++) {
+                        ((CapturesArguments) m).captureFrom(Array.get(rawArgument, i));
+                    }
+                } else {
+                    ((CapturesArguments) m).captureFrom(null);
```

Comment:
Should we capture null in case an null-array was passed? How can it be distinguished from an null-element? 
IMO it should be ignored cause `null` is strictly speaken an `<Type[]>` but the captor bound to `<Type>`.